bot/exts/backend/error_handler.py

```diff
@@ -15,6 +18,51 @@
 log = get_logger(__name__)
 
 
+class CheckedDeleteMessageButton(DeleteMessageButton):
+    """Button to delete a message, restricted to the message owner and moderators."""
+
+    def __init__(self, owner: User | Member):
+        super().__init__()
+        self.owner = owner
+
+    async def callback(self, interaction: Interaction) -> None:
+        """Delete the message if the user is the owner of the message or a moderator."""
+        user = interaction.user
+        if (
+            user.id == self.owner.id or
+            (
+                isinstance(user, Member) and
+                any(user.get_role(role) for role in MODERATION_ROLES)
```

Comment:
Out of scope for this PR probably, but we do this check in various places in the codebase. I wonder if there should be a single function that does it to un-ugly-fy it.

Reply:
Yeah, possibly something that could go in bot-core. I'll leave it out of this PR though

Reply:
Added https://github.com/python-discord/bot-core/pull/194 that can help with this, but doesn't need to be part of this PR.